radix-engine/src/system/system_modules/costing/fee_reserve.rs

```diff
@@ -386,6 +391,33 @@ impl SystemLoanFeeReserve {
         }
     }
 
+    pub fn is_deferred_costs_and_loan_covered(&self) -> Result<bool, FeeReserveError> {
+        let deferred_execution_cost = self
+            .effective_execution_cost_unit_price
+            .checked_mul(self.execution_cost_units_deferred)
+            .ok_or(FeeReserveError::Overflow)?;
+
+        let mut deferred_storage_cost = Decimal::ZERO;
+        for (t, size) in &self.storage_cost_deferred {
+            let cost = match t {
+                StorageType::State => self.state_storage_price,
+                StorageType::Archive => self.archive_storage_price,
+            }
+            .checked_mul(*size)
+            .ok_or(FeeReserveError::Overflow)?;
+            deferred_storage_cost = deferred_storage_cost
+                .checked_add(cost)
+                .ok_or(FeeReserveError::Overflow)?;
+        }
+
+        let sum = deferred_execution_cost
+            .checked_add(deferred_storage_cost)
+            .and_then(|x| x.checked_add(self.xrd_owed))
```

Comment:
I initially thought this was wrong, but I think this is right actually :D

Can we add a comment here explaining that `xrd_owed` starts as `system_loan_in_xrd` but changes to `0` after the loan is repaid in full :+1: